Modules/selectmodule.c

```diff
@@ -213,7 +213,7 @@ select_select(PyObject *self, PyObject *args)
         tvp = (struct timeval *)NULL;
     else {
         if (_PyTime_FromSecondsObject(&timeout, timeout_obj,
-                                      _PyTime_ROUND_CEILING) < 0) {
+                                      _PyTime_ROUND_UP) < 0) {
```

Comment:
Different direction of rounding shouldn't be mixed. Use `_PyTime_ROUND_UP` in `_PyTime_AsTimeval()` below, otherwise the timeout from -999 to -1 nanoseconds will be rounded to 0.

Reply:
Solved in 3f7ccda.

Reply:
Should I use `_PyTime_ROUND_UP` as well in:

```C
 _PyTime_AsTimeval_noraise(timeout, &tv, _PyTime_ROUND_CEILING); 
```? 

(line 285)

Reply:
No, this is not needed. At line 285 the value is not negative.

Reply:
:ok_hand:

Reply:
Thank you for the patience!